### PR TITLE
refactor: use parse_program() for parsing

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit removes "FileType" added in #445. 

This is done by removing "AstParser::parse_script" and "AstParser::parse_module"
in favor of "AstParser::parse_program" which uses new "parse_program"
API provided by SWC.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@magurotuna please review

File: src/rules/no_inner_declarations.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ use swc_ecmascript::ast;
 use swc_ecmascript::visit::noop_visit_type;
 use swc_ecmascript::visit::Node;
 use swc_ecmascript::visit::Visit;
+// use swc_ecmascript::visit::VisitAll;
```
Comment (magurotuna) [inline src/rules/no_inner_declarations.rs:9]: Maybe you forgot to delete this, or intended to replace `Visit` with `VisitAll`?

File: src/lib.rs [hunk 2]
```diff
@@ -80,6 +76,7 @@ mod lint_tests {
       false,
     );
 
+    eprintln!("diagnostics {:#?}", diagnostics);
```
Comment (magurotuna) [inline src/lib.rs:79]: Is this `println!` just for debugging?

**Review by magurotuna — COMMENTED** [review]
Just a few nitpicks, otherwise LGTM 👍

File: src/rules/no_inner_declarations.rs (hunk 1 above)
Comment (bartlomieju) [reply src/rules/no_inner_declarations.rs]: Forgot to delete

File: src/lib.rs (hunk 2 above)
Comment (bartlomieju) [reply src/lib.rs]: Yep, removed

**Review by magurotuna — APPROVED** [review]
LGTM

**Pull request merged**